minos/common/model/fields.py

```diff
@@ -41,16 +42,35 @@ def type(self) -> t.Type:
         """Type getter."""
         return self._type
 
+    @property
+    def parser(self) -> t.Optional[t.Callable[[t.Any], T]]:
+        """Parser getter."""
+        return self._parser
+
+    @property
+    def _parser_name(self) -> t.Optional[str]:
+        if self.parser is None:
+            return None
+        return self.parser.__name__
+
     @property
     def value(self) -> t.Any:
         """Value getter."""
         return self._value
 
     @value.setter
     def value(self, data: t.Any) -> t.NoReturn:
+        """Check if the given value is correct and stores it if ``True``, otherwise raises an exception.
+
+        :param data: new value.
+        :return: This method does not return anything.
         """
-        check if the value is correct
-        """
+        if self._parser is not None:
+            try:
+                data = self.parser(data)
+            except Exception as exc:
+                log.warning(f"Raised {repr(exc)} while parsing {repr(self.name)} field with {repr(data)} value.")
```

Comment:
Hi @andrea-mucci, should this be the expected behavior when an exception is raised while calling the corresponding parser? I mean, ignoring it and trying to continue without parsing, or should we propagate the exception?

Reply:
The Exception should be propagated.